### PR TITLE
Enable Microsoft.Extensions.Primitives

### PR DESCRIPTION
**Pull request by maryamariyan**
- Fix ref/src build on supported frameworks
- Use GenerateReferenceSource to properly update ref file
- Enable tests
- Add packaging setup

### CODE REVIEW

File: src/libraries/Directory.Build.props [hunk 1]
```diff
@@ -5,14 +5,19 @@
   <Import Project="..\..\Directory.Build.props" />
 
   <ItemGroup>
+    <PortedExtensionsProject Include="$(MSBuildThisFileDirectory)Microsoft.Extensions.Primitives\ref\**\*csproj" />
```
Comment (ericstj) [inline src/libraries/Directory.Build.props]: Can you collapse into a single item?

File: src/libraries/Microsoft.Extensions.Primitives/ref/Microsoft.Extensions.Primitives.cs [hunk 2]
```diff
@@ -34,52 +37,48 @@ public partial interface IChangeToken
         bool HasChanged { get; }
         System.IDisposable RegisterChangeCallback(System.Action<object> callback, object state);
     }
-    [System.Diagnostics.DebuggerDisplayAttribute("Value = {_value}")]
-    [System.ObsoleteAttribute("This type is obsolete and will be removed in a future version.")]
```
Comment (ericstj) [inline src/libraries/Microsoft.Extensions.Primitives/ref/Microsoft.Extensions.Primitives.cs]: Obsolete needs to remain

File: src/libraries/Microsoft.Extensions.Primitives/src/ThrowHelper.cs [hunk 3]
```diff
@@ -53,7 +53,7 @@ internal static ArgumentException GetArgumentException(ExceptionResource resourc
 
         private static string GetResourceText(ExceptionResource resource)
         {
-            return Resources.ResourceManager.GetString(GetResourceName(resource), Resources.Culture);
+            return SR.GetResourceString(GetResourceName(resource));
```
Comment (ericstj) [inline src/libraries/Microsoft.Extensions.Primitives/src/ThrowHelper.cs]: Can you spot check the behavior here to make sure this is working correctly?

**Review by ericstj — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Primitives/ref/Microsoft.Extensions.Primitives.cs (hunk 2 above)
Comment (maryamariyan) [reply src/libraries/Microsoft.Extensions.Primitives/ref/Microsoft.Extensions.Primitives.cs]: I can bring back the Obsolete, but this diff as-is was generated by GenerateReferenceSource target

Comment (ericstj) [reply src/libraries/Microsoft.Extensions.Primitives/ref/Microsoft.Extensions.Primitives.cs]: Understood.  Let me see about fixing that up by removing that suppression in a separate PR.

**Comment by ericstj** [comment]
Can you also add an SLN for this?

File: src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj [hunk 4]
```diff
@@ -4,6 +4,19 @@
     <TargetFrameworks>$(DefaultNetCoreTargetFramework);net472</TargetFrameworks>
```
Comment (ericstj) [inline src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj]: I think this needs to change to $(NetCoreAppCurrent).  Without it these tests aren't running.

File: src/libraries/Microsoft.Extensions.Primitives/src/ThrowHelper.cs (hunk 3 above)
Comment (maryamariyan) [reply src/libraries/Microsoft.Extensions.Primitives/src/ThrowHelper.cs]: tested locally, works correctly

**Comment by ericstj** [comment]
Awesome!  Small nit: in the future make sure to squash these (I know we merged the one when we were bringing history, so github probably changed your default).

**Pull request merged**